Cargo.toml

```diff
@@ -46,6 +47,13 @@ verbose = []
 "socket-tcp" = ["socket"]
 "socket-icmp" = ["socket"]
 "async" = []
+
+defmt-trace = []
+defmt-debug = []
+defmt-info = []
+defmt-warn = []
+defmt-error = []
+
```

Comment:
Hm, what's the point of these, are they needed by `defmt`?

Reply:
Yes, the defmt macros only emit code if the corresponding log level (or lower) is enabled. The end-user is supposed to depend on smoltcp with `defmt, defmt-debug` features to see up to `debug` level, for example. This is so log level is configurable per-crate.

Reply:
But don't you need to forward the features then? I'm confused about how this would work as-is in this PR.